projects/igniteui-angular/src/lib/core/styles/typography/_bootstrap.scss

```diff
@@ -96,7 +96,9 @@
     @include snackbar-typography();
     @include switch-typography();
     @include tabs-typography();
-    @include time-picker-typography();
+    @include time-picker-typography($categories: (
+        selected-time: 'h4'
```

Comment:
You have to list all typography styles related to the time-picker here, not only the one that you want to update. This way the 'header-hour' and the 'time-period' styles won't work in bootstrap theme.